services/api/src/api/routes/endpoint.py

```diff
@@ -51,60 +143,45 @@ def create_endpoint(
     max_age_short: int = 0,
 ) -> Endpoint:
     async def processing_step_endpoint(request: Request) -> Response:
-        if not processing_steps:
-            raise MissingProcessingStepsError("Missing processing steps or empty list")
-
-        processing_step = processing_steps[0]
-        # TODO: Add support to iterate over the processing steps list in case of failure
         try:
+            # validating request parameters
             dataset = request.query_params.get("dataset")
+
             if not are_valid_parameters([dataset]) or not dataset:
                 raise MissingRequiredParameterError("Parameter 'dataset' is required")
-            if processing_step.input_type == "dataset":
-                config = None
-                split = None
-            elif processing_step.input_type == "config":
-                config = request.query_params.get("config")
-                split = None
-                if not are_valid_parameters([config]):
-                    raise MissingRequiredParameterError("Parameter 'config' is required")
-            else:
-                config = request.query_params.get("config")
-                split = request.query_params.get("split")
-                if not are_valid_parameters([config, split]):
-                    raise MissingRequiredParameterError("Parameters 'config' and 'split' are required")
-            logging.info(f"{processing_step.job_type}, dataset={dataset}, config={config}, split={split}")
+
+            config = request.query_params.get("config")
+            split = request.query_params.get("split")
+
+            input_types = get_input_types_by_priority(steps_by_input_type=steps_by_input_type)
+
+            logging.debug(f"endpoint={endpoint_name} dataset={dataset} config={config} split={split}")
```

Comment:
it's a lot of if/elif... I'm wondering if we could simplify a bit (not sure how)

Reply:
I think a strategy pattern will work in this case, let me try